classroom/snippets/snippets.py

```diff
@@ -13,13 +13,110 @@
 # limitations under the License.
 
 from __future__ import print_function
+import json
 from googleapiclient import errors
 
 
 class ClassroomSnippets(object):
     def __init__(self, service):
         self.service = service
 
+    def create_course(self):
+        """
+        Creates a 10th Grade Biology course.
+        """
+        service = self.service
+        # [START classroom_create_course]
+        course = {
+            'name': '10th Grade Biology',
+            'section': 'Period 2',
+            'descriptionHeading': 'Welcome to 10th Grade Biology',
+            'description': """We'll be learning about about the
+                         structure of living creatures from a
+                         combination of textbooks, guest lectures,
+                         and lab work. Expect to be excited!""",
+            'room': '301',
+            'ownerId': 'me',
+            'courseState': 'PROVISIONED'
+        }
+        course = service.courses().create(body=course).execute()
+        print('Course created:', course.get('name'), course.get('id'))
+        # [END classroom_create_course]
+
+    def get_course(self):
+        """
+        Retrieves a classroom course by its id.
+        """
+        service = self.service
+        # [START classroom_get_course]
+        course_id = '123456'
+        try:
+            course = service.courses().get(id=course_id).execute()
+            print('Course "{%s}" found.', course.get('name'))
+        except errors.HttpError as e:
+            error = json.loads(e.content).get('error')
+            if error.get('code') == 404:
+                print('Course with ID "{%s}" not found.', course_id)
+            else:
+                raise
+        # [END classroom_get_course]
+
+    def list_courses(self):
+        """
+        Lists all classroom courses.
+        """
+        service = self.service
+        # [START classroom_list_courses]
+        courses = []
+        page_token = None
+
+        while True:
```

Comment:
Is it likely the caller will have several hundred courses? I'd actually just skip checking the page token do a regular list like the Drive Quickstart to keep the code snippet clean:
https://developers.google.com/drive/api/v3/quickstart/python

You could consider adding a pagination like the one Calendar:
https://developers.google.com/calendar/v3/pagination

Reply:
i think we show it this way because developers aren't always aware that the call only returns a set number so this makes it a little clearer?